homeassistant/components/tesla_fleet/climate.py

```diff
@@ -0,0 +1,309 @@
+
+
+COP_MODES = {
+    "Off": HVACMode.OFF,
+    "On": HVACMode.COOL,
+    "FanOnly": HVACMode.FAN_ONLY,
+}
+
+# String to celsius
+COP_LEVELS = {
+    "Low": 30,
+    "Medium": 35,
+    "High": 40,
+}
+
+# Celsius to IntEnum
+TEMP_LEVELS = {
+    30: CabinOverheatProtectionTemp.LOW,
+    35: CabinOverheatProtectionTemp.MEDIUM,
+    40: CabinOverheatProtectionTemp.HIGH,
+}
+
+
+class TeslaFleetCabinOverheatProtectionEntity(TeslaFleetVehicleEntity, ClimateEntity):
+    """Tesla Fleet vehicle cabin overheat protection entity."""
+
+    _attr_precision = PRECISION_WHOLE
+    _attr_target_temperature_step = 5
+    _attr_min_temp = COP_LEVELS["Low"]
+    _attr_max_temp = COP_LEVELS["High"]
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = list(COP_MODES.values())
+    _enable_turn_on_off_backwards_compatibility = False
+    _attr_entity_registry_enabled_default = False
+
+    def __init__(
+        self,
+        data: TeslaFleetVehicleData,
+        scopes: Scope,
+    ) -> None:
+        """Initialize the cabin overheat climate entity."""
+
+        # Scopes
+        self.read_only = Scope.VEHICLE_CMDS not in scopes or data.signing
+
+        super().__init__(data, "climate_state_cabin_overheat_protection")
+
+        # Supported Features
+        if self.read_only:
+            self._attr_supported_features = ClimateEntityFeature(0)
+        else:
+            self._attr_supported_features = (
+                ClimateEntityFeature.TURN_ON | ClimateEntityFeature.TURN_OFF
+            )
+            if self.get("vehicle_config_cop_user_set_temp_supported"):
+                self._attr_supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the entity."""
+
+        if (state := self.get("climate_state_cabin_overheat_protection")) is None:
+            self._attr_hvac_mode = None
+        else:
+            self._attr_hvac_mode = COP_MODES.get(state)
+
+        # If not scoped, prevent the user from changing the HVAC mode by making it the only option
+        if self._attr_hvac_mode and self.read_only:
+            self._attr_hvac_modes = [self._attr_hvac_mode]
+
+        if (level := self.get("climate_state_cop_activation_temperature")) is None:
+            self._attr_target_temperature = None
+        else:
+            self._attr_target_temperature = COP_LEVELS.get(level)
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+        await self.async_set_hvac_mode(HVACMode.COOL)
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+        await self.async_set_hvac_mode(HVACMode.OFF)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+
+        if not (temp := kwargs.get(ATTR_TEMPERATURE)):
+            return
```

Comment:
should this be a validation error or is it expected?

Reply:
This was based on a bad assumption, I'll change this (and the other `async_set_temperature` to raise a ServiceValidationError if no temperature is provided.

Reply:
I think the schema prohibits this from being empty? Then this check is unnecessary.

Reply:
You can call the service with a high and low temp even if the integraton doesnt support that feature.

Reply:
I noticed after making my comment so I have made a PR to implement this validation directly in the `ClimateEntity` so integrations doesn't need to check.